src/main/java/io/warp10/ext/interpolation/InterpolatorLinear.java

```diff
@@ -0,0 +1,193 @@
+//
+//   Copyright 2022  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.ext.interpolation;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptLib;
+import io.warp10.script.WarpScriptMapperFunction;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import org.apache.commons.math3.analysis.interpolation.AkimaSplineInterpolator;
+import org.apache.commons.math3.analysis.interpolation.LinearInterpolator;
+import org.apache.commons.math3.analysis.interpolation.SplineInterpolator;
+import org.apache.commons.math3.analysis.polynomials.PolynomialSplineFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Function that implements the bicubic spline interpolation
+ */
+public class InterpolatorLinear extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public static enum TYPE {
+    LINEAR,SPLINE, AKIMA
+  };
+  
+  private TYPE type;
+  
+  private static class LINEAR extends NamedWarpScriptFunction implements WarpScriptStackFunction, WarpScriptMapperFunction {
+
+    private final PolynomialSplineFunction func;
+    private final String generatedFrom;
+    private ArrayList xval;
+    private ArrayList fval;
+
+    private LINEAR(PolynomialSplineFunction function, String interpolatorName, ArrayList xval, ArrayList fval) {
+      super(interpolatorName);
+      func = function;
+      this.xval = xval;
+      this.fval = fval;
+      generatedFrom = interpolatorName;
+    }
+
+    private double value(double x) {
+      if (!func.isValidPoint(x)) {
+        return Double.NaN;
+      } else {
+        return func.value(x);
+      }
+    }
+
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object o = stack.pop();
+      if (!(o instanceof Number)) {
+        throw new WarpScriptException(getName() + " expects a DOUBLE or a LONG");
+      }
+
+      double x = ((Number) o).doubleValue();
+      stack.push(value(x));
+
+      return stack;
+    }
+
+    @Override
+    public Object apply(Object[] args) throws WarpScriptException {
+      long tick = (long) args[0];
+      long[] locations = (long[]) args[4];
+      long[] elevations = (long[]) args[5];
+      Object[] values = (Object[]) args[6];
+
+      if (1 != values.length) {
```

Comment:
also need to handle case 0 == values.length separately for empty buckets

Reply:
Not relevant here since function is applied on values and not on ticks.
Applying on tick as a mapper will be supported with a future PR